src/Razor/src/Microsoft.VisualStudio.RazorExtension/RazorPackage.cs

```diff
@@ -3,25 +3,25 @@
 
 using System;
 using System.Runtime.InteropServices;
+using System.Threading;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
 
 namespace Microsoft.VisualStudio.RazorExtension
 {
-    [PackageRegistration(UseManagedResourcesOnly = true)]
+    [PackageRegistration(UseManagedResourcesOnly = true, AllowsBackgroundLoading = true)]
     [AboutDialogInfo(PackageGuidString, "ASP.NET Core Razor Language Services", "#110", "#112", IconResourceID = "#400")]
     [Guid(PackageGuidString)]
-    public sealed class RazorPackage : Package
+    public sealed class RazorPackage : AsyncPackage
     {
         public const string PackageGuidString = "13b72f58-279e-49e0-a56d-296be02f0805";
 
         private const string CSharpPackageIdString = "13c3bbb4-f18f-4111-9f54-a0fb010d9194";
 
-        protected override void Initialize()
+        protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
-            base.Initialize();
-
-            ThreadHelper.ThrowIfNotOnUIThread();
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
So we make this async, then Allow it to load in the Background, but then we force it onto the Main/UI thread anyway? I assume this is something about their extension model?

Reply:
I think we can probably just remove all of this. 

But to answer @ryanbrandenburg -s question, yeah that's what's happening. Most things in VS traditionally will (or need to) happen on the UI thread. `SwitchToMainThreadAsync` is a construct for asynchronously switching to the main thread (without blocking the thread).

Reply:
Awesomeness, will remove it allz